src/main/java/gregtech/common/covers/filter/FluidFilterContainer.java

```diff
@@ -1,128 +1,175 @@
 package gregtech.common.covers.filter;
 
-import gregtech.api.gui.GuiTextures;
-import gregtech.api.gui.Widget;
-import gregtech.api.gui.widgets.LabelWidget;
-import gregtech.api.gui.widgets.SlotWidget;
+import gregtech.api.mui.GTGuiTextures;
 import gregtech.api.util.IDirtyNotifiable;
 
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
 import net.minecraftforge.common.util.INBTSerializable;
 import net.minecraftforge.fluids.FluidStack;
-import net.minecraftforge.items.ItemStackHandler;
 
+import com.cleanroommc.modularui.api.drawable.IKey;
+import com.cleanroommc.modularui.api.widget.IWidget;
+import com.cleanroommc.modularui.api.widget.Interactable;
+import com.cleanroommc.modularui.screen.ModularPanel;
+import com.cleanroommc.modularui.utils.Alignment;
+import com.cleanroommc.modularui.value.sync.GuiSyncManager;
+import com.cleanroommc.modularui.value.sync.PanelSyncHandler;
+import com.cleanroommc.modularui.value.sync.SyncHandlers;
+import com.cleanroommc.modularui.widgets.ButtonWidget;
+import com.cleanroommc.modularui.widgets.ItemSlot;
+import com.cleanroommc.modularui.widgets.layout.Row;
 import org.jetbrains.annotations.NotNull;
 
+import java.io.IOException;
+import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
-import java.util.function.Supplier;
 
-public class FluidFilterContainer implements INBTSerializable<NBTTagCompound> {
+public class FluidFilterContainer extends BaseFilterContainer<FluidStack, FluidFilter>
+                                  implements INBTSerializable<NBTTagCompound> {
 
-    private final ItemStackHandler filterInventory;
     private final FluidFilterWrapper filterWrapper;
 
-    public FluidFilterContainer(IDirtyNotifiable dirtyNotifiable, int capacity) {
-        this.filterWrapper = new FluidFilterWrapper(dirtyNotifiable, capacity);
-        this.filterInventory = new ItemStackHandler(1) {
-
-            @Override
-            public boolean isItemValid(int slot, @NotNull ItemStack stack) {
-                return FilterTypeRegistry.getFluidFilterForStack(stack) != null;
-            }
+    public FluidFilterContainer(IDirtyNotifiable dirtyNotifiable) {
+        super(dirtyNotifiable);
+        this.filterWrapper = new FluidFilterWrapper(this); // for compat
```

Comment:
Can you explain how this keeps compat exactly? I think it's probably best to fully remove the old filter code in this case, otherwise the new design will likely become an abstraction over the old instead of a proper rework.

Reply:
someone was using filterwrappers for something, so i was thinking of keeping the class around for a while and route the old methods back to the filter container.

Reply:
Gt breaks compat every update anyways. Just yeet it

Reply:
yeeted